### PR TITLE
build(i18n): Drop target support for languages

### PR DESCRIPTION
**Pull request by scinos**
#### Background

Now that we have dropped the `fallback` build, it doesn't make sense to generate language files in `public/evergreen/languages` and `public/fallback/languages`. It doesn't even make sense to have a `target` anymore.

This PR removes all notion of `target` form the build-languages scripts and generated files.

#### Changes proposed in this Pull Request

* Move translations from `public/<target>/languages` to just `public/languages`
* Move `./calypso-strings.pot` to `./public/calypso-strings.pot`. I think it is better to keep all build artifacts inside `public` and don't pollute the root.
* Move `./chunks-map.<target>.json` to `./public/chunks-map.json`. Ditto.
* Refactor `./bin/build-languages.js` to use `node-fetch` instead of https streams. Streams were getting me lots of `ECONNRESET` problems, and now that we don't need to save a downloaded file in several places, they don't provide any huge benefit over the Fetch API. 
* Refactor `./bin/build-languages.js` to just use `console.log` instead of readline to manipulate stdout. The readline logic was messing with debugging in VSCode and made the build logs hard to follow in Docker.

#### Testing instructions

* Open calypso.live link, go to your profile and change the language. You should see a bunch of `<locale>-.json` files loaded in the Network tab every time you go to a different page.
* Go to gutenboarding (add `/new` to the calypso.live link) and check you can change languages using the top-right lang selector.

### CODE REVIEW

File: bin/build-languages.js [hunk 1]
```diff
@@ -1,48 +1,24 @@
...
 const _ = require( 'lodash' );
 const mkdirp = require( 'mkdirp' );
+const fetch = require( 'node-fetch' );
 
 const LANGUAGES_BASE_URL = 'https://widgets.wp.com/languages/calypso';
 const LANGUAGES_REVISIONS_FILENAME = 'lang-revisions.json';
-const CALYPSO_STRINGS = './calypso-strings.pot';
-const CHUNKS_MAP_PATTERN = './chunks-map.*.json';
+const CALYPSO_STRINGS = './public/calypso-strings.pot';
+const CHUNKS_MAP_PATTERN = './public/chunks-map.json';
```
Comment (yuliyan) [inline bin/build-languages.js:14]: I like the idea of keeping the root directory clean and I suppose there's no concern for these to be publicly accessible.

**Review by yuliyan — DISMISSED** [review]
Really nice clean up!

I think the [translations related gitignore rules](https://github.com/Automattic/wp-calypso/blob/897c7c3639619339b0d879b69d2612d70ce67929/.gitignore#L58-L59) can be dropped as well.

~~Apart from that, it LGTM 🚢~~ Seems like dev environment with enabled translation chunks fails to boot. See https://github.com/Automattic/wp-calypso/pull/56396#issuecomment-924156896.

**Comment by scinos** [comment]
> I think the translations related gitignore rules can be dropped as well. Apart from that, it LGTM 🚢

I'm in two minds about dropping them. Every developer that ever run `yarn start` with translation chunks enabled will have those files in the filesystem, so the moment we drop the gitignore rule they will pop up in the next PR.

But if we don't drop them, those ignore rules are going to be in `.gitignore` forever. Although is not a huge problem, I'm pretty sure some of the entries haven't applied for months/years anyway 🤷🏼

**Comment by yuliyan** [comment]
I've decided to test booting in a clean development environment with enabled translations chunks, i.e. `yarn run clean && BUILD_TRANSLATION_CHUNKS=true yarn start`, and I noticed that it fails due to missing `chunks-map.json` during build languages script, which should normally be the case.

After some digging, I found that build languages script was being executed twice - first time from the [`build-client-if-prod` script](https://github.com/Automattic/wp-calypso/blob/dd19114b0012b17ead5a9721cc25f5ccc0c87cbb/package.json#L80) before webpack build has finished, and second time from the [server bundler](https://github.com/Automattic/wp-calypso/blob/dd19114b0012b17ead5a9721cc25f5ccc0c87cbb/client/server/bundler/index.js#L36).

In this case, the former is unwanted behaviour caused by a missing parentheses around `(yarn run build-client && yarn run build-languages-if-enabled)` in `build-client-if-prod`. This mistake remained unnoticed since I've originally implemented it, because previously the build languages script was looping through the existing chunk-map.json files, so in this case, since chunk maps were missing during the first run, the chunks build step was being skipped.

**Comment by scinos** [comment]
> I found that build languages script was being executed twice

Very good catch!!! Fixed with `{}` to avoid creating a subshell (I found it in https://www.gnu.org/software/bash/manual/html_node/Command-Grouping.html}

**Review by yuliyan — APPROVED** [review]
Tested again and it LGTM 🚢

---

> Fixed with {} to avoid creating a subshell (I found it in https://www.gnu.org/software/bash/manual/html_node/Command-Grouping.html}

TIL 👍
